docs/parser_plugins.md

```diff
@@ -1 +1,210 @@
+This has lead to five distinct categories of responsibility for the developer to manage:
+
+1. select the relevant files and read them into Python.
+2. target and extract relevant data fields within the source file format on a per-file basis.
+3. match the data of interest with their counterparts in the target/NOMAD schema.
+4. reshape and mangle the data to match the target/NOMAD `Quantity`s' specification. This may include computing derived properties not present in the original source files.
+5. build up a Python `archive` object using the classes provided by the target/NOMAD schema.
+
+In the past, we encountered cases where the delineation between these responsibilities blurred, resulting in widely different parser designs.
+Especially when dealing with the larger, more feature-rich parsers, these designs could become quite complex.
+
+During refactoring, we therefore addressed these responsibilities on a more individual basis.
+This lead to a clear defined _interfaces_ and powerful _tools_ for the developer.
+These are further complimented by _best practices_ and _design protocols_.
+
+More specifically, point 1 (file selection) is now handled by the `MatchingParser` class, from which the parser class inherits. It coordinates with the NOMAD base to scan the uploaded folder tree and select the relevant files.
+
+Point 4 (data mangling) is covered by the NOMAD schema, as you saw in the part [NOMAD-Simulations schema](nomad_simulations.md).
+Not only does this allow for a more consistent handling of the data, it also cleanly partitions the work between parser and schema developers.
+In principle, a parser should at most perform trivial manipulations, such as slicing an array or multiplying two extracted values.
+More on the exact "how" later.
+
+Lastly, we will cover tools for point 5 (building the `archive`) and 2 (targeting data fields) in the next subsections.
+Point 5, in particular, was a developing nuisance, as it required the reconstruction of the already defined target/NOMAD schema.
+That comes across as doing double work: running over the schema once to define it, and a second time to initialize it.
+The main hurdle to automation here was the variable structure of the source data:
+a well-defined hierarchical format does not ensure that individual file structures remains constant.
+Some leaf nodes or branches may simply not be present.
+
+With all of these points covered, developing a parser now comes down to just implementing point 3 (matching source with target).
+This is the stage where data actually becomes semantically enriched and standardized.
+This is a profoundly human task (barring any further leaps in LLM capabilities), where domain expertise is most needed.
+It is thus this scientific curation task that is now most emphasized in parser development, rather than the coding aspects.
+
+Let us cover the remaining tools, so we can move on to the real science.
+
+### Parsing Hierarchical Tree Formats
```

Comment:
At this point the reader still has no idea how anything in the plugin works in practice. You mentioned the template but we haven't worked with it at all yet. We need some section/s before this which explain the basics before you start talking about hierarchical data trees, like how to link nomad-simulations and import the schema and populate a quantity in the most basic way.

Reply:
I was thinking of adding the folder structure once the template is set up. Would that address your concerns?

Reply:
I think that the new additions now show enough of the folder structure and parser code to provide a clear view.
Will resolve this thread, but feel free to open it if you disagree.